flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/DataStreamSink.java

```diff
@@ -180,6 +180,18 @@ public DataStreamSink<T> setParallelism(int parallelism) {
         return this;
     }
 
+    /**
+     * Sets the max parallelism for this sink. The degree must be higher than zero and less than the
+     * upper bound.
```

Comment:
> The degree must be higher than zero and less than the upper bound.

Just stating this in java doc is not safe enough, we'd better have directly sanity check for this. For example:
`OperatorValidationUtils.validateMaxParallelism(maxParallelism, yes);`